src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'rectangle';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
 export class Triangle {
+  shape: Shape = 'triangle';
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || a + b <= c
+      || b + c <= a
+      || a + c <= b
+    ) {
+      throw new Error('side of triangle is wrong');
+    }
```

Comment:
It would be great to explain what exactly is wrong with the side of the triangle, what the user should write to satisfy the condition.